src/readDICOMTags.js

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable-next-line no-unused-vars */
+import regeneratorRuntime from 'regenerator-runtime'
+
+import createWebworkerPromise from './createWebworkerPromise'
+import PromiseFileReader from 'promise-file-reader'
+import WorkerPool from './WorkerPool'
+
+import config from './itkConfig'
+
+const workerFunction = async (
+  webWorker,
+  fileDescriptions,
+  tags
+) => {
+  let worker = webWorker
+  const { webworkerPromise, worker: usedWorker } = await createWebworkerPromise(
+    'ImageIO',
+    worker
+  )
+  worker = usedWorker
+
+  const transferables = fileDescriptions.map(description => {
+    return description.data
+  })
+  const message = {
+    operation: 'readDICOMTags',
+    fileDescriptions: fileDescriptions,
+    tags,
+    config
+  }
+  const tagResults = await webworkerPromise.postMessage(message, transferables)
+  return { tags: tagResults, webWorker: worker }
+}
+const numberOfWorkers = navigator.hardwareConcurrency ? navigator.hardwareConcurrency : 4
+const workerPool = new WorkerPool(numberOfWorkers, workerFunction)
+
+const readDICOMTags = async (
+  fileList,
+  tags = null
```

Comment:
These parameters may change, pending API discussions.
- fileList: probably going to become a single file rather than a list of files
- tags: if we want a null/undefined `tags` arg to mean "give me all tags", then this will stay. Right now there is no logic for getting all tags.

Reply:
+1 for the single file. Specification of the tags-per-file is simpler this way, too. +1 for the current `null` returns all tags API.